sunflower-land/sunflower-land
# [FEAT] Deposit Bumpkin

# Description

This PR includes the flows to support depositing a Bumpkin into the game.

https://github.com/sunflower-land/sunflower-land/discussions/2591

The main benefits of this include:

- Instant equip items without a transaction
- Perform Wearable Drops in game = no withdrawal fees
- Require in game items to mint clothing

## Workflow

When a player visits the game, they will need to follow the prompts to deposit their Bumpkin onto the farm. When you deposit a Bumpkin, all of the wearables that are currently equipped will also be sent to the farm.

Players can also deposit wearables from the standard deposit flows.

To withdraw, players go to the Bank and either click "Withdraw Wearables" or "Withdraw Bumpkin". When a Bumpkin is withdrawn, the wearables that are equipped will be sent back to the Bumpkin's Wallet.

A player cannot withdraw any wearables that are currently in use by the Bumpkin. They must first dequip the Bumpkin.

## Screenshots

Ignore the shadow NPCs - this is a local issue

<img width="532" alt="Screen Shot 2023-05-31 at 12 56 02 pm" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/4ba4c3c2-b1b2-4081-8359-4a00086d2160">
<img width="522" alt="Screen Shot 2023-05-31 at 12 57 59 pm" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/3a2cbe19-ac46-4db2-ace6-3f1e3c31425a">
<img width="552" alt="Screen Shot 2023-05-31 at 12 57 50 pm" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/f776a9d4-bdfb-4fef-9f31-e307e596d7ef">
<img width="526" alt="Screen Shot 2023-05-31 at 12 57 42 pm" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/6d54a719-2e8f-4366-ba3f-1ab8fb63082a">
<img width="534" alt="Screen Shot 2023-05-31 at 1 00 11 pm" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/f6b1ede2-f29d-431a-b28f-d66323883c59">

<img width="525" alt="Screen Shot 2023-05-31 at 1 00 56 pm" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/e1d25c50-083b-47dc-ae0c-2d61e0f883c1">

## How to test?

Run locally pointing at the API.

Make sure you include these in your `.env`

```
# Contracts (Testnet)
VITE_ACCOUNT_MINTER_CONTRACT=0xEA2EAa3DD7814D9034B2db40274b353B4b27d3AB
VITE_SESSION_CONTRACT=0x10Af5c3D3316b66f2219717C7AB3F900a9E9f706
VITE_GAME_CONTRACT=0x294223b4dBD0635b0E3354DDD3aaa4792efa590B
VITE_WITHDRAWAL_CONTRACT=0x204d1457B743673095Ba253f447aA0f11669ED47
VITE_DEPOSIT_CONTRACT=0x5228045162c5058A6b19CBC6563B16D231f432Fa
```

**Launch Plan 🚀 **

This is a highly anticipated feature, but due to the architecture changes involved we will need thorough testing. We will likely hijack Testnet for a few days to test this feature properly.

This will also require a lot of notice to current players to prepare them for the change.
